### PR TITLE
Save Profile Button Fadesout While Saving

### LINKED ISSUE #3871: Delay after clicking Save button and actual save needs some kind of loading indicator
### Steps
1. Go to staging.osf.io
2. Go to one of your account settings
3. Change any field in your Profile Information
4. Click the Save button
5. Notice the delay between clicking Save and the green "Settings updated" text appearing

### Expected behavior
I expected that while there's a delay between clicking Save and actually saving, I would get some indication that my changes were being processed, such as the Save button's text changing into a lighter color "Saving" button until the "Settings updated" message appeared below.

For example, the Add button in the New wiki page modal changes to "Creating New Wiki page" before the page is successfully created.
screen shot 2015-07-29 at 9 58 29 am

In project settings, the link to connect/import an account changes to a grayed out "Loading..." message before the account is successfully imported.

screen shot 2015-07-29 at 2 46 47 pm

### Actual behavior

save

Since there's a delay before the "Settings updated" message appears, the user may think the changes were successful and try to leave the page before they are actually processed.

### PR DESCRIPTION
**Pull request by Johnetordoff**
<h1> Purpose </h1>
Indicate the save profile button is saving. Close #3871.
<h1> Changes </h1>
Adds observable so to indicates saving and changes the mako files to express these changes.
<h1> Side Effects </h1>
None that I know of.

### CODE REVIEW

File: website/static/js/profile.js [hunk 1]
```diff
@@ -207,6 +209,15 @@ var BaseViewModel = function(urls, modes, preventUnsaved) {
     // Must be set after isValid is defined in inherited view models
     self.hasValidProperty = ko.observable(false);
 
+    self.savingText = ko.computed(function() {
```
Comment (lyndsysimon) [inline website/static/js/profile.js]: This seems to shadow the declaration of `self.savingText()` on [Line 198](https://github.com/CenterForOpenScience/osf.io/pull/3941/files#diff-9cd61d5035427d03e2bdefab7288fd55R198).

File: website/static/js/profile.js [hunk 2]
```diff
@@ -207,6 +209,15 @@ var BaseViewModel = function(urls, modes, preventUnsaved) {
...
+        if(self.saving()){
```
Comment (lyndsysimon) [inline website/static/js/profile.js]: This conditional could be simplified using a ternary operator:

```javascript
return self.saving() ? 'Saving' : 'Save';
```

File: website/static/js/profile.js [hunk 3]
```diff
@@ -327,8 +339,11 @@ BaseViewModel.prototype.cancel = function(data, event) {
 
 };
 
+
 BaseViewModel.prototype.submit = function() {
-    if (this.hasValidProperty() && this.isValid()) {
+
+    this.saving(true);
+   if (this.hasValidProperty() && this.isValid()) {
```
Comment (lyndsysimon) [inline website/static/js/profile.js]: The indentation level of this line does not match that of the previous.

File: website/static/js/profile.js [hunk 4]
```diff
@@ -336,13 +351,14 @@ BaseViewModel.prototype.submit = function() {
             this.handleSuccess.bind(this)
         ).done(
             this.setOriginal.bind(this)
+        ).done(
```
Comment (lyndsysimon) [inline website/static/js/profile.js]: This `done` callback is empty.

File: website/templates/include/profile/jobs.mako [hunk 5]
```diff
@@ -112,7 +112,8 @@
                 <button
                         type="submit"
                         class="btn btn-success"
-                    >Save</button>
+                        data-bind="text: savingText(), enable : !saving()">
```
Comment (lyndsysimon) [inline website/templates/include/profile/jobs.mako]: I believe you can bind on both `enable` and `disable`. Binding to `disable` would avoid negating the result of `saving()`, making it slightly more readable.

This is entirely subjective, so as a wise man once told me in a code review - follow your heart :heart:.

**Comment by sloria** [comment]
Closing this for now.

**Pull request closed without merging**